src/core/qgsfield.cpp

```diff
@@ -208,8 +209,30 @@ QString QgsField::displayString( const QVariant &v ) const
     return QgsApplication::nullRepresentation();
   }
 
-  if ( d->type == QVariant::Double && d->precision > 0 )
-    return QString::number( v.toDouble(), 'f', d->precision );
+  if ( d->type == QVariant::Double )
+  {
+    if ( d->precision > 0 )
+    {
+      return QLocale().toString( v.toDouble(), 'f', d->precision );
+    }
+    else
+    {
+      // Precision is not set, let's guess it from the
```

Comment:
I'm concerned about the performance implications of this block - could we bypass it for "normal" locales and just return v.toString() instead?

Reply:
Do you mean "." == "normal"?

Reply:
Yeah, of course ;) Comma is just weird.

Reply:
@nyalldawson I don't think we can: if we go for `toString` we'll miss the thousands separator (we get `123456789.0123` instead of `123'456'789.0123`  (or whatever the system's locale settings say it should be). If I'm not wrong, the range widget formatter does that already and (among other issues) this PR tries to make the numbers look the same whether the range widget of the plain text widget is used.

Do you have any reason to believe that it would make a noticeable performance difference?

Reply:
could we bypass this for "normal" locales where "normal" is defined as `.` for decimal separator and nothing for the thousands separator?

Reply:
Well, yes, but if we do, we'll loose the thousand grouping for `.` locales and we get `123456789.0123` instead of `123'456'789.0123` (or whatever the system's locale settings say it should be).
```
In [8]: QLocale(QLocale.German).toString( 123456789.012345, 'f', 6 )
Out[8]: '123.456.789,012345'

In [9]: QLocale(QLocale.English).toString( 123456789.012345, 'f', 6 )
Out[9]: '123,456,789.012345'
```

Reply:
I think you are missing the second condition of my definition of normal.

Reply:
Ah, yes, sorry. Ok then.